generators/app/index.js

```diff
@@ -384,6 +394,105 @@ module.exports = class extends Generator {
                 });
             },
 
+            askForPublisherName: () => {
+                return generator.prompt({
+                    type: 'input',
+                    name: 'publisherName',
+                    message: 'What name would you like to publish this extension under?',
+                }).then(publisherAnswer => {
+                    generator.extensionConfig.publisherName = publisherAnswer.publisherName;
+                });
+            },
+
+            askForExistingNotebooks: () => {
+                if (generator.extensionConfig.type !== 'ext-notebook') {
+                    return Promise.resolve();
+                }
+
+                return generator.prompt({
+                    type: 'confirm',
+                    name: 'addNotebooks',
+                    message: 'Add existing notebooks to be shipped?',
+                    default: true
+                }).then(existingNotebook => {
+                    generator.extensionConfig.addNotebooks = existingNotebook.addNotebooks;
+                });
+            },
+
+            askForNotebooks: () => {
+                if (generator.extensionConfig.type !== 'ext-notebook') {
+                    return Promise.resolve();
+                }
+
+                if (generator.extensionConfig.addNotebooks) {
+                    return generator.prompt({
+                        type: 'input',
+                        name: 'notebookPath',
+                        message: 'Provide the absolute path to the folder containing your notebooks.',
+                        default: '/Desktop/notebooks'
```

Comment:
this path can be wrong in mac or linux

Reply:
should I even provide a default path? wasn't sure because it might be helpful to provide an example but could be confusing for different OS

Reply:
default path is good. I was just worried that the path won't work in mac. /Desktop is not a correct path in mac or linux. you can use environment variables to get the home path in Windows, linux and mac so you don't hard code it

---
In reply to: [464657100](https://github.com/llali/generator-azuredatastudio/pull/23#discussion_r464657100) [](ancestors = 464657100)